core/base/src/main/java/org/eclipse/dataspaceconnector/core/HttpClientExtension.java

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.core;
+
+import okhttp3.EventListener;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Response;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.system.Inject;
+import org.eclipse.dataspaceconnector.spi.system.Provider;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+import static java.util.Optional.ofNullable;
+
+public class HttpClientExtension implements ServiceExtension {
+
+    @EdcSetting(value = "If true, enable HTTPS call enforcement. Default value is 'false'", type = "boolean")
+    public static final String EDC_HTTP_ENFORCE_HTTPS = "edc.http.enforce-https";
+
+    /**
+     * An optional OkHttp {@link EventListener} that can be used to instrument OkHttp client for collecting metrics.
+     * Used by the optional {@code micrometer} module.
+     */
+    @Inject(required = false)
+    private EventListener okHttpEventListener;
+
+    @Override
+    public String name() {
+        return "HTTP client";
+    }
+
+    @Provider
+    public OkHttpClient addHttpClient(ServiceExtensionContext context) {
+        var builder = new OkHttpClient.Builder()
+                .connectTimeout(30, TimeUnit.SECONDS)
+                .readTimeout(30, TimeUnit.SECONDS);
+
+        ofNullable(okHttpEventListener).ifPresent(builder::eventListener);
+
+        boolean enforceHttps = context.getSetting(EDC_HTTP_ENFORCE_HTTPS, false);
+        if (enforceHttps) {
+            builder.addInterceptor(new EnforceHttps());
+        } else {
+            context.getMonitor().info("HTTPS enforcement it's not enabled, it should be in a production environment");
```

Comment:
```suggestion
            context.getMonitor().info("HTTPS enforcement it not enabled, please do not use this in a production environment");
```

Reply:
@paullatzelsperger the double negation make it pretty confusing, what about "please enable it in a production environment"?

Reply:
strictly speaking not a double negative (that would be e.g. "don't not use it..."), but your suggestion is fine too :)